### PR TITLE
nodejs week2

### PR DESCRIPTION
**Pull request by patel-prakashkumar**
I have completed nodejs week2 Assignment.

### CODE REVIEW

File: nodejs/week2/app.js [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+const app = express();
+const data = require("./document.json");
+const port = process.env.PORT || 3000;
+app.use(express.json())
+app.get('/', (req, res) => {
+    res.send('This is a search engine')
+})
+/* 
+app.get('/search', (req, res) => {
+ const searchWord = req.query.q
```
Comment (itsgauravsaxena) [inline nodejs/week2/app.js]: Please remove commented out code. 
Its called zomby code https://zombiecodekill.com/2014/01/09/zombie-code/

File: nodejs/week2/app.js [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+    const foundId = data.find(x => x.id === parseInt(req.params.id));
+    if (!foundId) {
+        res.status(404).json("ID is not Availble")
+    }
+    else {
+        res.json(foundId)
+    }
+});
+app.post('/search', (req, res) => {
+    const listItem = req.body.fields;
```
Comment (itsgauravsaxena) [inline nodejs/week2/app.js:33]: Search should also accept params and then either if only param is provided, it should allow search. And if both params and body is provided, thwn it should throw an error.

File: nodejs/week2/app.js [hunk 3]
```diff
@@ -0,0 +1,67 @@
...
+    const listFoundItem = data.filter((item) => {
+        return Object.keys(listItem).some(key => {
+            return item[key] === listItem[key];
+        });
+    });
+    console.log(listFoundItem)
```
Comment (itsgauravsaxena) [inline nodejs/week2/app.js]: Nice to have - console log should be removed.

**Review by itsgauravsaxena — CHANGES_REQUESTED** [review]

File: nodejs/week2/app.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+    else{
+        res.json(listItem)
+    }
+}) */
+
+app.get('/search', (req, res) => {
+    const searchWord = req.query.q
+    const listItem = data.filter(e => Object.values(e).some(value => String(value).includes(searchWord)))
+    if (listItem.length === 0) {
+        res.status(404).json("Search Word is not availbel in Data")
```
Comment (itsgauravsaxena) [inline nodejs/week2/app.js]: According to exercise, if no query params are provided, then it should return all the records in json file.

**Review by itsgauravsaxena — CHANGES_REQUESTED** [review]

File: nodejs/week2/app.js (hunk 3 above)
Comment (Shahnaz200) [reply nodejs/week2/app.js]: good job

**Comment by Shahnaz200** [comment]
nice work!

File: nodejs/week2/app.js (hunk 1 above)
Comment (patel-prakashkumar) [reply nodejs/week2/app.js]: yah... I will remove commented out code..

File: nodejs/week2/app.js (hunk 2 above)
Comment (patel-prakashkumar) [reply nodejs/week2/app.js:33]: ok.. Its seem to be wrong logic.  I will update code.

File: nodejs/week2/app.js (hunk 3 above)
Comment (patel-prakashkumar) [reply nodejs/week2/app.js]: yah.. I just remove from code. Thanks for feedback.

File: nodejs/week2/app.js (hunk 4 above)
Comment (patel-prakashkumar) [reply nodejs/week2/app.js]: Yahh. you are right.. I will update code.

**Pull request merged**